### DISCUSSION TITLE
Fetch submission success an error messages with GraphQL

### DISCUSSION THREAD

**Discussion by Stalex89** [solspace/craft-freeform#734] [Questions & Answers] [answered]
Hello,

Sorry for duplicating [this post](https://github.com/solspace/craft-freeform/discussions/214) but it has still no answer. Could you please clarify how to fetch submission success an error messages with GraphQL?

![Screenshot 2023-07-13 at 11 46 08](https://github.com/solspace/craft-freeform/assets/23526237/dc443442-614d-41ee-b387-9ea10d0cf061)

I cannot find GraphQL example of it in documentation.

Thank you in advance!

**Comment by kjmartens** [comment, marked as answer]
Hi @Stalex89,

This should already be possible. If you have a look here, you'll see that it is included in the example JSON response:
https://docs.solspace.com/craft/freeform/v4/headless/graphql/#example-json-response

``` json
"class": "",
"method": "post",
"enctype": "application\/x-www-form-urlencoded",
"successMessage": "Form has been submitted successfully!",
"errorMessage": "Sorry, there was an error submitting the form. Please try again.",
"honeypot": {
    "name": "freeform_form_handle",
    "value": ""
},
```

Please see the [How to Render a Form guide](https://docs.solspace.com/craft/freeform/v4/headless/graphql/#how-to-render-a-form) for more information. 🙂 

However, I see that we're not exposing them via GraphQL, so I've made a note about this and we'll get them added and available in a release next week. 🙂
